src/main/java/org/folio/rest/service/StreamService.java

```diff
@@ -66,17 +68,22 @@ public String enhanceFlux(String firstFluxId, Flux<String> secondFlux, String co
     while(firstString != null || secondString != null) {
```

Comment:
Can this be done without the null checks by conditioning on hasNext? That would avoid many of the ternaries in this block of code, and perhaps be a little more straight forward.

Reply:
Unfortunately I don't think we can. The null checks are done after the iterators are advanced, and they don't always both advance at the same time.